src/openms/include/OpenMS/FORMAT/OSWFile.h

```diff
@@ -61,8 +64,36 @@ namespace OpenMS
   */
   class OPENMS_DLLAPI OSWFile
   {
-public:
-  
+  public:
+    
+    /// query all proteins, not just one with a particular ID
+    static constexpr Size ALL_PROTEINS = -1;
+
+    OSWFile(const String& filename);
+    OSWFile(const OSWFile& rhs) = default;
+    OSWFile& operator=(const OSWFile& rhs) = default;
+
+    /// read data from an SQLLite OSW file into @p swath_result
+    /// Depending on the number of proteins, this could take a while. If you want a faster
+    /// method, use readMinimal().
+    void read(OSWData& swath_result);
+
+    /// reads in transitions and a list of protein names/IDs
+    /// but no peptide/feature/transition mapping data (which could be very expensive)
+    /// Use in conjunction with on-demand readProtein() to query proteins as needed
+    void readMinimal(OSWData& swath_result);
+
+    /**
+      @brief populates a protein at index @p index with Peptides, unless the protein already has peptides
+
+      Internally uses the proteins ID to search for cross referencing peptides and transitions in the OSW file.
+      @throws Exception::InvalidValue if the ID is not present in the OSW file
+    */
+    void readProtein(OSWData& swath_result, const Size index);
```

Comment:
is this efficient to individually query them?

Reply:
32 sec for a full query and 44 if done individually.
The thing is, I don't want to let the user wait 44 secs in TOPPView when annotating, so I just read the protein list and fill them in on demand (very fast for single items)

Reply:
ok that's faster than expected. maybe add this motivation as a comment?